From a829cb5f89a1da5fa33f67350fa26267de29f8ec Mon Sep 17 00:00:00 2001
From: Vicentiu Galanopulo <vicentiu@balena.io>
Date: Thu, 7 Oct 2021 15:28:30 +0200
Subject: [PATCH] Set fec1 to fixed-link mode

Set the fec1 ethernet controller into fixed-link mode
since it will not be connected to a PHY anymore but directly
to the KSZ9893R switch

Upstream-status: Inappropriate [configuration]
Signed-off-by: Vicentiu Galanopulo <vicentiu@balena.io>
---
 arch/arm64/boot/dts/compulab/cl-som-imx8.dts | 14 +++-----------
 1 file changed, 3 insertions(+), 11 deletions(-)

diff --git a/arch/arm64/boot/dts/compulab/cl-som-imx8.dts b/arch/arm64/boot/dts/compulab/cl-som-imx8.dts
index 2afa1fcb8495..bbed1d8399d4 100644
--- a/arch/arm64/boot/dts/compulab/cl-som-imx8.dts
+++ b/arch/arm64/boot/dts/compulab/cl-som-imx8.dts
@@ -482,20 +482,12 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_fec1>;
 	phy-mode = "rgmii-id";
-	phy-handle = <&ethphy0>;
 	fsl,magic-packet;
 	status = "okay";
 
-	mdio {
-		#address-cells = <1>;
-		#size-cells = <0>;
-
-		ethphy0: ethernet-phy@0 {
-			compatible = "ethernet-phy-ieee802.3-c22";
-			reg = <0>;
-			at803x,led-act-blind-workaround;
-			at803x,eee-disabled;
-		};
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
 	};
 };
 
-- 
2.17.1

